### PR TITLE
Add deployments and auth config needed for IAP on Jupyterhub 

### PR DESCRIPTION
**Pull request by chiayi**
This adds the config and deployments that Jupyterhub will use for auth.

### CODE REVIEW

File: jupyter-on-gke/iap_module/deployments/backend-config.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+apiVersion: cloud.google.com/v1
+kind: BackendConfig
+metadata:
+  name: config-default
+  namespace: ray
+spec:
+  iap:
+    enabled: true
+    oauthclientCredentials:
+      secretName: my-secret
```
Comment (imreddy13) [inline jupyter-on-gke/iap_module/deployments/backend-config.yaml:24]: Where is this created?

File: jupyter-on-gke/jupyter_config/config-selfauth.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      description: "Creates GPU VMs (T4) as the compute for notebook execution"
+      kubespawner_override:
+        image: jupyter/tensorflow-notebook:python-3.10
+        extra_resource_limits:
+          # number of gpus needed on the node
+          nvidia.com/gpu: "2"
+      #  possible values: nvidia-tesla-k80, nvidia-tesla-p100, nvidia-tesla-p4, nvidia-tesla-v100, nvidia-tesla-t4, nvidia-tesla-a100, nvidia-a100-80gb, nvidia-l4
+        node_selector:
+          cloud.google.com/gke-accelerator: "nvidia-tesla-t4"
+    - display_name: "GPU T4"
```
Comment (imreddy13) [inline jupyter-on-gke/jupyter_config/config-selfauth.yaml:89]: What's the difference between these? Should one be A100 instead?

File: jupyter-on-gke/iap_module/deployments/managed-cert.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+
+apiVersion: networking.gke.io/v1
+kind: ManagedCertificate
+metadata:
+  name: managed-cert
+  namespace: ray
+spec:
+  domains:
+    - ${ip_addr}
```
Comment (imreddy13) [inline jupyter-on-gke/iap_module/deployments/managed-cert.yaml:22]: How does this get set?

File: jupyter-on-gke/iap_module/deployments/backend-config.yaml (hunk 1 above)
Comment (chiayi) [reply jupyter-on-gke/iap_module/deployments/backend-config.yaml:24]: The deployments will be created by Terraform and the variables will be passed in my Terraform. The TF files will be in the next PR.

File: jupyter-on-gke/jupyter_config/config-selfauth.yaml (hunk 2 above)
Comment (chiayi) [reply jupyter-on-gke/jupyter_config/config-selfauth.yaml]: Fixed. We should also make sure that the platform will crate a100 gpu nodes as well for standard right?

File: jupyter-on-gke/iap_module/deployments/managed-cert.yaml (hunk 3 above)
Comment (chiayi) [reply jupyter-on-gke/iap_module/deployments/managed-cert.yaml:22]: This will be passed in by the tf files.

**Review by imreddy13 — APPROVED** [review]

**Pull request merged**